File: englising-be/src/main/java/org/englising/com/englisingbe/word/entity/TrackWord.java
package org.englising.com.englisingbe.word.entity;

import jakarta.persistence.*;
import lombok.*;
import org.englising.com.englisingbe.music.entity.Lyric;
import org.englising.com.englisingbe.music.entity.Track;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.sql.Timestamp;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TrackWord {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "track_word_id")
    private Long trackWordId;

    @ManyToOne
    @JoinColumn(name = "track_id", referencedColumnName = "track_id")
    private Track track;

    @ManyToOne
    @JoinColumn(name = "lyric_id", referencedColumnName = "lyric_id")
    private Lyric lyric;

    @ManyToOne
    @JoinColumn(name = "word_id", referencedColumnName = "word_id")
    private Word word;

    @Column(name = "word_index")
    private Integer wordIndex;

    @Column(name = "origin_word")
    private String originWord;

    @CreationTimestamp
    @Column(name = "created_at")
    private Timestamp createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private Timestamp updatedAt;
}

File: englising-recommend/model/track_word.py
from sqlalchemy import Column, ForeignKey, Integer, String, Float, Boolean, TIMESTAMP, DECIMAL, Text, Date, Table
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from model.base import Base


class TrackWord(Base):
    __tablename__ = 'track_word'
    track_word_id = Column(Integer, primary_key=True)
    track_id = Column(Integer, ForeignKey('track.track_id'), nullable=False)
    lyric_id = Column(Integer, ForeignKey('lyric.lyric_id'), nullable=False)
    word_id = Column(Integer, ForeignKey('word.word_id'), nullable=False)
    word_index = Column(Integer, nullable=False)
    origin_word = Column(String, nullable=False)
    importance = Column(Float, nullable=True)
    created_at = Column(TIMESTAMP, default=func.now())
    updated_at = Column(TIMESTAMP, onupdate=func.now())

    track = relationship("Track", back_populates="track_words")
    lyric = relationship("Lyric", back_populates="track_words")
    word = relationship("Word", back_populates="track_words")
